lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
+    attempts = 10
+    loop do
+      result = get_attributes(path,false)
+      # return false for file not found
+      return false if result == INVALID_FILE_ATTRIBUTES
+      # return true if we found a file/dir and it's not a symlink
+      return true if (result & FILE_ATTRIBUTE_REPARSE_POINT) != FILE_ATTRIBUTE_REPARSE_POINT
+      # walk the symlink and try again...
+      path = readlink(path)
```

Comment:
So I just verified that walking the links is how we're supposed to behave, and how we document this functioning.  Based on the code changes, it seems like a change in behavior... and I was going to suggest the following tests at line 246 of `file_system_spec.rb`:

```ruby
it "should return true for exist? when resolving a symlink chain pointing to a file" do
      # point symlink -> file
      symlink = tmpfile("somefile_link")
      Puppet::FileSystem.symlink(file, symlink)

      # point symlink2 -> symlink
      symlink2 = tmpfile("somefile_link2")
      Puppet::FileSystem.symlink(symlink, symlink2)

      expect(Puppet::FileSystem.exist?(symlink2)).to be_truthy
    end

    it "should return false for exist? when resolving a symlink chain that dangles" do
      # point symlink -> file
      symlink = tmpfile("somefile_link")
      Puppet::FileSystem.symlink(file, symlink)

      # point symlink2 -> symlink
      symlink2 = tmpfile("somefile_link2")
      Puppet::FileSystem.symlink(symlink, symlink2)

      # yank file, and make symlink dangle
      ::File.delete(file)

      # symlink2 is now indirectly dangled
      expect(Puppet::FileSystem.exist?(symlink2)).to be_falsey
    end
```
What's weird is that these tests pass before and after your changes, but I haven't debugged in to understand why.